### ISSUE TITLE
"Subquery returns more than 1 row" error with MySQL persistence layer

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#953]
This error happens on publish, in cases where the `orbeon_form_definition` table contains multiple instances of the same form, with different `created` timestamps, but the same `last_modified` timestamp. For instance:

``` sql
select *
from orbeon_form_definition
where
    app =  'orbeon'
    and form =  'empty'
    and last_modified = '2013-01-23 15:47:26';
```

Returns:

![Screen Shot 2013-04-22 at 11 16 54 AM](https://f.cloud.github.com/assets/106961/410777/da6fa404-ab78-11e2-9cbd-d04940a62e20.png)

It is unclear when and why this happens. A workaround consists in deleting that specific published form from the `orbeon_form_definition` table:

``` sql
delete 
from orbeon_form_definition
where
    app =  'orbeon'
    and form =  'empty';
```

**Comment by ebruchez** [comment]
@avenet is this issue still relevant with the latest changes to the relational persistence layer?

**Comment by avernet** [comment]
@ebruchez Yes, this issue still relevant: in 2016.2 we introduced a unique id (`data_id`), but this id is only used by the indexes, so this issue can still happen. But adding this id is a good step towards fixing this.

**Comment by ebruchez** [comment]
@avernet So what would be the next steps?

**Comment by avernet** [comment]
@ebruchez We should change all the queries to use `last_modified` somehow to instead use or join on `orbeon_i_current`. This will also make the queries simpler, less error prone, and faster.

**Comment by ebruchez** [comment]
@avernet Got it, and noting after our discussion to check all the places where we use `max(last_modified_time)`.

**Comment by ebruchez** [comment]
Added evaluation for consideration for 2016.3 pending bug review.
